File: 4.SumOfPrimes/SieveEratosthenes.py
import math

def eratosthenes(num):
	sum = 0
	isComposite = []
	upperBoundSQRT = int(math.sqrt(num))
	for indx in range(0, num):
		isComposite.append(False)
	for x in range(2, upperBoundSQRT):
		if(isComposite[x] == False):
			for k in range(x*x, num, x):
				isComposite[k] = True;
	for x in range(2, num):
		if(isComposite[x] == False):
			sum = sum + x
	return sum

print eratosthenes(2000000)

File: reference/SieveEratosthenes.cpp
#include <math.h>
#include <iostream>

using namespace std;

void runEratosthenesSieve(int upperBound) {
      long sum = 0;
      int upperBoundSquareRoot = (int)sqrt((double)upperBound);
      bool *isComposite = new bool[upperBound + 1];
      for(int i = 0; i < (upperBound + 1); i++){
            isComposite[i] = false;
      }
      for (int m = 2; m <= upperBoundSquareRoot; m++) {
            if (!isComposite[m]) {
                  for (int k = m * m ; k <= upperBound; k += m){
                        cout << "k is " << k << endl;
                        isComposite[k] = true;
                  }
            }
      }
      for (int m = 2; m <= upperBound; m++)
            if (!isComposite[m])
                  sum += m;
      delete [] isComposite;
      cout << sum << endl;

}

int main()
{
      int max = 100;
      runEratosthenesSieve(max);
}